scripts/zkai_attention_kv_d32_two_head_seq32_fused_softmax_table_native_gate.py

```
@@ -0,0 +1,1113 @@
+    ) as handle:
+        tmp_path = pathlib.Path(handle.name)
+        writer = csv.DictWriter(
+            handle, fieldnames=TSV_COLUMNS, delimiter="\t", lineterminator="\n"
+        )
+        writer.writeheader()
+        writer.writerow(row)
+    try:
+        with tmp_path.open("r", encoding="utf-8", newline="") as handle:
+            rows = list(csv.DictReader(handle, delimiter="\t"))
+        if rows != [expected_row]:
+            raise AttentionKvD32TwoHeadSeq32FusedSoftmaxTableGateError("TSV round-trip drift")
```

Ruff (0.15.13)

[warning] Abstract `raise` to an inner function

(TRY301)